ui.apps/src/main/javascript/minicart/src/utils/cartInitializer.js

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+import { useEffect } from 'react';
+import { useCartState } from './state';
+import { useCookieValue } from './hooks';
+import { useMutation } from '@apollo/react-hooks';
+
+import MUTATION_CREATE_CART from '../queries/mutation_create_guest_cart.graphql';
+import CART_DETAILS_QUERY from '../queries/query_cart_details.graphql';
+import MUTATION_REMOVE_ITEM from '../queries/mutation_remove_item.graphql';
+import MUTATION_ADD_TO_CART from '../queries/mutation_add_to_cart.graphql';
+
+const CartInitializer = props => {
+    const [{ cartId: stateCartId }, dispatch] = useCartState();
+    const CART_COOKIE = 'cif.cart';
+
+    const [cartId, setCartCookie] = useCookieValue(CART_COOKIE);
+    const [createCart, { data, error }] = useMutation(MUTATION_CREATE_CART);
+    const [addItem] = useMutation(MUTATION_ADD_TO_CART);
+    const [removeItem] = useMutation(MUTATION_REMOVE_ITEM);
+
+    const _createFunctions = (cartId, dispatch) => {
+        return {
```

Comment:
What's the purpose of this function? Couldn't we achieve the same result by using `useCallback`? 
```javascript
removeItem = useCallback( (itemId) => { ... }, 
[cartId, dispatch])
```

Reply:
As discussed in slack, we keep the renamed method to handle two different cases for creating the cart handler functions :)